# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/TableCellAccessible.h
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/generic/TableCellAccessible.h
--- a/accessible/generic/TableCellAccessible.h	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/generic/TableCellAccessible.h	Wed Mar 23 04:00:26 2022 +0000
@@ -7,59 +7,22 @@
 #ifndef mozilla_a11y_TableCellAccessible_h__
 #define mozilla_a11y_TableCellAccessible_h__
 
-#include "nsTArray.h"
-#include <stdint.h>
+#include "mozilla/a11y/TableCellAccessibleBase.h"
+#include "TableAccessible.h"
 
 namespace mozilla {
 namespace a11y {
 
 class LocalAccessible;
-class TableAccessible;
 
 /**
- * Abstract interface implemented by table cell accessibles.
+ * Base class for LocalAccessible table cell implementations.
  */
-class TableCellAccessible {
+class TableCellAccessible : public TableCellAccessibleBase {
  public:
-  /**
-   * Return the table this cell is in.
-   */
-  virtual TableAccessible* Table() const = 0;
-
-  /**
-   * Return the column of the table this cell is in.
-   */
-  virtual uint32_t ColIdx() const = 0;
-
-  /**
-   * Return the row of the table this cell is in.
-   */
-  virtual uint32_t RowIdx() const = 0;
-
-  /**
-   * Return the column extent of this cell.
-   */
-  virtual uint32_t ColExtent() const { return 1; }
-
-  /**
-   * Return the row extent of this cell.
-   */
-  virtual uint32_t RowExtent() const { return 1; }
-
-  /**
-   * Return the column header cells for this cell.
-   */
-  virtual void ColHeaderCells(nsTArray<LocalAccessible*>* aCells);
-
-  /**
-   * Return the row header cells for this cell.
-   */
-  virtual void RowHeaderCells(nsTArray<LocalAccessible*>* aCells);
-
-  /**
-   * Returns true if this cell is selected.
-   */
-  virtual bool Selected() = 0;
+  virtual TableAccessible* Table() const override = 0;
+  virtual void ColHeaderCells(nsTArray<Accessible*>* aCells) override;
+  virtual void RowHeaderCells(nsTArray<Accessible*>* aCells) override;
 
  private:
   LocalAccessible* PrevColHeader();